cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -83,6 +91,92 @@ public void download(String url, Path target) {
     }
   }
 
+  /**
+   * Downloads a file while showing a progress bar
+   *
+   * @param url the url to download
+   * @param target Path of the target directory
+   * @param response {@link HttpResponse} to use
+   */
+  private void downloadFileWithProgressBar(String url, Path target, HttpResponse<InputStream> response) throws IOException {
+
+    String contentLength = response.headers().firstValue("content-length").orElse("undefined");
+    if (contentLength.equals("undefined")) {
+      throw new IllegalStateException("Content-Length was not provided by download source: " + url);
+    }
+
+    long size = Long.parseLong(contentLength);
+
+    byte[] data = new byte[1024];
+    boolean fileComplete = false;
+    int count;
```

Comment:
To be pedantic, a `long` would be the correct value type here instead of `int`.
I do not really assume that IDEasy will ever download files bigger than 2 terra-bytes but I want to teach developers to avoid accidentally creating such unnecessary limitations. In a customer project a 2 TB limit could cause a production bug one day :)

Reply:
Sorry, my mistake. This is not the counter of the total filesize but only the size of a chunk that is anyway limited to the array lenth (1024).
Can you avoid redundantly specifying the buffer size `1024` as literal (use `data.length` instead of repeating 1024).